src/renderer.rs

```diff
@@ -637,25 +655,34 @@ impl Renderer {
                                                  u_transform_array);
                         }
                         DrawCommandInfo::Composite(ref info) => {
-                            let needs_fb = match info.blend_mode {
-                                MixBlendMode::Normal => unreachable!(),
-
-                                MixBlendMode::Screen |
-                                MixBlendMode::Overlay |
-                                MixBlendMode::ColorDodge |
-                                MixBlendMode::ColorBurn |
-                                MixBlendMode::HardLight |
-                                MixBlendMode::SoftLight |
-                                MixBlendMode::Difference |
-                                MixBlendMode::Exclusion |
-                                MixBlendMode::Hue |
-                                MixBlendMode::Saturation |
-                                MixBlendMode::Color |
-                                MixBlendMode::Luminosity => true,
-
-                                MixBlendMode::Multiply |
-                                MixBlendMode::Darken |
-                                MixBlendMode::Lighten => false,
+                            let needs_fb = match info.operation {
+                                CompositionOp::MixBlend(MixBlendMode::Normal) => unreachable!(),
+
+                                CompositionOp::MixBlend(MixBlendMode::Screen) |
+                                CompositionOp::MixBlend(MixBlendMode::Overlay) |
+                                CompositionOp::MixBlend(MixBlendMode::ColorDodge) |
+                                CompositionOp::MixBlend(MixBlendMode::ColorBurn) |
+                                CompositionOp::MixBlend(MixBlendMode::HardLight) |
+                                CompositionOp::MixBlend(MixBlendMode::SoftLight) |
+                                CompositionOp::MixBlend(MixBlendMode::Difference) |
+                                CompositionOp::MixBlend(MixBlendMode::Exclusion) |
+                                CompositionOp::MixBlend(MixBlendMode::Hue) |
+                                CompositionOp::MixBlend(MixBlendMode::Saturation) |
+                                CompositionOp::MixBlend(MixBlendMode::Color) |
+                                CompositionOp::MixBlend(MixBlendMode::Luminosity) |
+                                CompositionOp::Filter(LowLevelFilterOp::Blur(..)) |
+                                CompositionOp::Filter(LowLevelFilterOp::Brightness(_)) |
+                                CompositionOp::Filter(LowLevelFilterOp::Contrast(_)) |
+                                CompositionOp::Filter(LowLevelFilterOp::Grayscale(_)) |
+                                CompositionOp::Filter(LowLevelFilterOp::HueRotate(_)) |
+                                CompositionOp::Filter(LowLevelFilterOp::Invert(_)) |
+                                CompositionOp::Filter(LowLevelFilterOp::Opacity(_)) |
```

Comment:
Can't opacity (and perhaps some of the others) be done without grabbing the framebuffer, just as a normal opengl blend op?

(We can still merge it if that's the case - but let's add a TODO noting that there is an optimization available for some of these modes to use the fixed function blend ops).